recognition/advanced_tree_search.py

```diff
@@ -536,6 +537,151 @@ def create_config(
         return config, post_config, lm_gc_job
 
 
+class RescoreLatticeCacheJob(rasr.RasrCommand, Job):
+    def __init__(
+        self,
+        crp,
+        lattice_cache,
+        rescorer_type="single-best",
+        max_hypotheses=5,
+        pruning_threshold=16.0,
+        history_limit=0,
+        rescoring_lookahead_scale=1.0,
+        rtf: float = 1.0,
+        cpu: int = 1,
+        mem: float = 4.0,
+        use_gpu: bool = False,
+        extra_config=None,
+        extra_post_config=None,
+    ):
+        self.set_vis_name("Rescore Lattice")
+
+        kwargs = locals()
+        del kwargs["self"]
+
+        (
+            self.config,
+            self.post_config,
+        ) = RescoreLatticeCacheJob.create_config(**kwargs)
+        self.input_lattice_cache = lattice_cache
```

Comment:
This Job assumes a single lattice cache as input but produces multiple caches as output.
Would the application then be to provide the bundle file for the parameter `lattice_cache`? 
And this is implemented because lattice processing would be done with fewer nodes than lattice generation?